### DISCUSSION TITLE
Tax categories, rates and prices

### DISCUSSION THREAD

**Discussion by skid** [vendurehq/vendure#970] [Q&A] [answered]
I need some clarification on how it all works:

My understanding: A `ProductVariant` can be assigned a `TaxCategory`, which in its turn can have different `TaxRates` depending on the `Zone`. If so, I don't understand how the Zone is decided, since it's needed obtain the tax rate and the price with tax. For example, I added 2 rates for the same TaxCategory and Zone and product variants (in the admin panel) decided to use the second rate. There is no way to control this from the UI it seems.

Additionally:

What's the difference between `ProductVariant.listPrice`, `ProductVariant.price`, `ProductVariant.productVariantPrices` - can these be controlled from the admin UI?

Is the price you input for the `ProductVariant` in the admin UI with tax included? (On my current deployment it looks like it is, since the shopAPI returns that price for `priceWithTax`)

**Comment by michaelbromley** [comment, marked as answer]
Hey! I'll try to answer each part:

> If so, I don't understand how the Zone is decided, since it's needed obtain the tax rate and the price with tax.

This is determined by the [TaxZoneStrategy](https://www.vendure.io/docs/typescript-api/tax/tax-zone-strategy/). By default, Vendure will just choose the default tax zone set on the current Channel: https://github.com/vendure-ecommerce/vendure/blob/3717192c30a9a4bb28975bbc3f219284b8e31d59/packages/core/src/config/tax/default-tax-zone-strategy.ts#L12-L16

> For example, I added 2 rates for the same TaxCategory and Zone and product variants (in the admin panel) decided to use the second rate

This is unexpected input. There should only be 1 rate defined per category/zone combination. Obviously this is not being enforced currently so that could be improved.

> What's the difference between ProductVariant.listPrice, ProductVariant.price, ProductVariant.productVariantPrices - can these be controlled from the admin UI?

* `listPrice` is the price as entered in the Admin UI in the variant detail view. This may be with or without tax included, depending on your tax setting (see later answer).
* `price` is the price **excluding tax**. This is a calculated property which depends on the current channel and tax settings.
* `productVariantPrices` is an array of prices, one for each Channel to which the ProductVariant is assigned. At run-time the `ProductVariantService.applyChannelPriceAndTax` [method](https://github.com/vendure-ecommerce/vendure/blob/a8ca0197dee97dfe7298353be4ddafc666434397/packages/core/src/service/services/product-variant.service.ts#L554) will select the correct one for the current channel and populate those values into the variant's `listPrice`, `listPriceIncludesTax`, and `currencyCode` fields.

> Is the price you input for the ProductVariant in the admin UI with tax included?

This is configurable. At the Channel level, you can specify whether "Prices include tax for the default Zone". 

![image](https://user-images.githubusercontent.com/6275952/124560945-a48ce300-de3d-11eb-890a-06d1509b7582.png)

More info here: https://www.vendure.io/docs/developer-guide/taxes/

**Reply by skid** [reply]
Great answer. Thank you
